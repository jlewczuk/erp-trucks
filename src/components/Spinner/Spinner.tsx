import styled from "styled-components";
import { rotate } from "../../helpers";

interface SpinnerOverlayProps {
  position?: "fixed" | "relative" | "absolute";
}

const SpinnerOverlay = styled.div<SpinnerOverlayProps>`
  height: 100%;
  width: 100%;
  display: flex;
  justify-content: center;
  align-items: center;
  background-color: rgba(0, 0, 0, 0.4);
  z-index: var(--top-z-index);
  position: ${(props) => props.position || "fixed"};
  ${(props) => props.position || "fixed"};
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
`;

const SpinnerContainer = styled.div`
  display: inline-block;
  width: 50px;
  height: 50px;
  border: 3px solid rgba(195, 195, 195, 0.6);
  border-radius: 50%;
  border-top-color: #636767;
  animation: ${rotate} 0.8s linear infinite;
`;

export const Spinner = ({ position }: SpinnerOverlayProps) => {
  return (
    <SpinnerOverlay position={position} data-testid="spinner-overlay">
      <SpinnerContainer />
    </SpinnerOverlay>
  );
};
