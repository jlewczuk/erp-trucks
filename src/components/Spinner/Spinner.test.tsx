import { Spinner } from "./Spinner";
import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";

describe("Spinner component", () => {
  test("it renders", () => {
    render(<Spinner />);
    expect(screen.getByTestId("spinner-overlay")).toBeInTheDocument();
  });

  test("it applies the right position", () => {
    render(<Spinner position="absolute" />);
    const spinnerOverlay = screen.getByTestId("spinner-overlay");
    expect(spinnerOverlay).toHaveStyle("position: absolute");
  });

  test("it matches snapshot", () => {
    const { asFragment } = render(<Spinner />);
    expect(asFragment()).toMatchSnapshot();
  });
});
