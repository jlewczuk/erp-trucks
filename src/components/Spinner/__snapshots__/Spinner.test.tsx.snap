// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinner component it matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-aYaIB ioSHbt"
    data-testid="spinner-overlay"
  >
    <div
      class="sc-gEvDqW dlMaxF"
    />
  </div>
</DocumentFragment>
`;
